src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -1,41 +1,100 @@
-import javax.servlet.http.HttpServletResponse;
-
-import com.google.codeu.data.Datastore;
-import com.google.gson.JsonObject;
-
-/**
- * Handles fetching site statistics.
- */
-@WebServlet("/stats")
-public class StatsPageServlet extends HttpServlet{
-
-  private Datastore datastore;
-
-  @Override
-  public void init() {
-    datastore = new Datastore();
-  }
-
-  /**
-   * Responds with site statistics in JSON.
-   */
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-
-    response.setContentType("application/json");
-
-    int messageCount = datastore.getTotalMessageCount();
-
-    JsonObject jsonObject = new JsonObject();
-    jsonObject.addProperty("messageCount", messageCount);
-    response.getOutputStream().println(jsonObject.toString());
-  }
-}
\ No newline at end of file
+package com.google.codeu.servlets;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.codeu.data.Datastore;
+import com.google.gson.JsonObject;
+
+/**
+ * Handles fetching site statistics.
+ */
+@WebServlet("/stats")
+public class StatsPageServlet extends HttpServlet {
+
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Responds with site statistics in JSON.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    response.setContentType("application/json");
+
+    // List of all messages
+    List<com.google.codeu.data.Message> allmessages = datastore.getAllMessages();
+
+    // Json for the JS
+    JsonObject jsonObject = new JsonObject();
+
+    // Number of messages
+    int messageCount = datastore.getTotalMessageCount();
+
+    // Number of messages property added to the Json
+    jsonObject.addProperty("messageCount", messageCount);
+
+    // Store of the biggest message
+    com.google.codeu.data.Message biggest = ((allmessages.size() == 0) ? null : allmessages.get(0));
+
+    /*
+     * countM, total number of messages users, register of all users
+     */
+    int countM = 0;
+    HashMap<String, Integer> usersAndNumOfMessages = new HashMap<>();
+
+    String userMostMessages = "";
+    int mostMessages = 0;
+
+    for (int i = 0; i < allmessages.size(); i++) {
+      // count for the avg of all messages
+      countM += allmessages.get(i).getText().length();
+
+      // getting the biggest message
+      if (allmessages.get(i).getText().length() > biggest.getText().length()) {
+        biggest = allmessages.get(i);
+      }
+
+
+      // getting and saving the user and message count and checking who has the most sent messages
+      if (!usersAndNumOfMessages.containsKey(allmessages.get(i).getUser())) {

```

Comment:
nit: Consider using a local variable to store allmessage.get(i).getUser() here so that you can use it here and everywhere below. I think that would make the code slightly easier to follow.